amethyst_assets/src/asset.rs

```diff
@@ -38,10 +38,80 @@ pub trait Format<A: Asset>: Send + 'static {
     type Options: Send + 'static;
 
     /// Reads the given bytes and produces asset data.
+    ///
+    /// ## Reload
+    ///
+    /// The reload structure has metadata which allows the asset management
+    /// to reload assets if necessary (for hot reloading).
+    /// You should only create this if `create_reload` is `true`.
+    /// Also, the parameter is just a request, which means you can also return `None`.
     fn import(
         &self,
         name: String,
         source: Arc<Source>,
         options: Self::Options,
-    ) -> Result<A::Data, BoxedErr>;
+        create_reload: bool,
+    ) -> Result<FormatValue<A>, BoxedErr>;
+}
+
+/// The `Ok` return value of `Format::import` for a given asset type `A`.
+pub struct FormatValue<A: Asset> {
+    /// The format data.
+    pub data: A::Data,
+    /// An optional reload structure
+    pub reload: Option<Box<Reload<A>>>,
+}
+
+impl<A: Asset> FormatValue<A> {
+    /// Creates a `FormatValue` from only the data (setting `reload` to `None`).
+    pub fn data(data: A::Data) -> Self {
+        FormatValue { data, reload: None }
+    }
+}
+
+/// This is a simplified version of `Format`, which doesn't give you as much as freedom,
+/// but in return is simpler to implement.
+/// All `SimpleFormat` types automatically implement `Format`.
+/// This format assumes that the asset name is the full path and also the only file.
```

Comment:
Minor grammar nit: Maybe this should end with "and the asset is only contained within one file."

Reply:
Makes sense, thanks.